File: app/src/main/java/com/islam/basepropject/ui/ExRecyclerView/Adapter.java
package com.islam.basepropject.ui.ExRecyclerView;

import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.islam.basepropject.R;
import com.islam.basepropject.project_base.base.adapters.BaseAdapter;
import com.islam.basepropject.project_base.base.adapters.BaseViewHolder;

public class Adapter extends BaseAdapter<String, Adapter.ViewHolder> {


    public Adapter() {
        super();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(parent, R.layout.item_string);
    }

    public class ViewHolder extends BaseViewHolder<String> {
        TextView textView;

        public ViewHolder(ViewGroup viewGroup, int layoutId) {
            super(viewGroup, layoutId);
            textView = itemView.findViewById(R.id.textView2);
        }

        @Override
        public void onBind(String item) {
            textView.setText(item);
        }
    }
}

File: app/src/main/java/com/islam/basepropject/ui/ExRecyclerView/Adapter.kt
package com.islam.basepropject.ui.ExRecyclerView

import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import com.islam.basepropject.R
import com.islam.basepropject.project_base.base.adapters.BaseListAdapter
import com.islam.basepropject.project_base.base.adapters.BaseViewHolder
import kotlinx.android.synthetic.main.item_string.*


class Adapter : BaseListAdapter<String, Adapter.ViewHolder>(object : DiffUtil.ItemCallback<String>() {
    override fun areItemsTheSame(oldItem: String, newItem: String): Boolean {
        return oldItem == newItem
    }

    override fun areContentsTheSame(oldItem: String, newItem: String): Boolean {
        return oldItem == newItem
    }
}) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(parent, R.layout.item_string)
    }

    inner class ViewHolder(viewGroup: ViewGroup, layoutId: Int) : BaseViewHolder<String>(viewGroup, layoutId) {
        override fun onBind(item: String) {
            textView2.text = item
        }
    }
}